airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -0,0 +1,797 @@
+
+    :param cluster_name: The unique name to give to your Amazon EKS Cluster.
+    :type cluster_name: str
+    :param cluster_role_arn: The Amazon Resource Name (ARN) of the IAM role that provides permissions for the
+       Kubernetes control plane to make calls to AWS API operations on your behalf.
+    :type cluster_role_arn: str
+    :param resources_vpc_config: The VPC configuration used by the cluster control plane.
+    :type resources_vpc_config: Dict
+    :param compute: The type of compute architecture to generate along with the cluster.
+        Defaults to 'nodegroup' to generate an EKS Managed Nodegroup.
+    :type compute: str
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+         If this is None or empty then the default boto3 behaviour is used. If
+         running Airflow in a distributed manner and aws_conn_id is None or
+         empty, then the default boto3 configuration would be used (and must be
+         maintained on each worker node).
+    :type aws_conn_id: str
+    :param region: Which AWS region the connection should use.
+        If this is None or empty then the default boto3 behaviour is used.
+    :type region: str
+
+    If compute is assigned the value of ``nodegroup``, the following are required:
+
+    :param nodegroup_name: The unique name to give your EKS Managed Nodegroup.
+    :type nodegroup_name: str
+    :param nodegroup_role_arn: The Amazon Resource Name (ARN) of the IAM role to associate
+         with the EKS Managed Nodegroup.
+    :type nodegroup_role_arn: str
+
+    """
+
+    template_fields = (
+        "cluster_name",
+        "cluster_role_arn",
+        "resources_vpc_config",
+        "nodegroup_name",
+        "nodegroup_role_arn",
+        "compute",
+        "aws_conn_id",
+        "region",
+    )
+
+    def __init__(
+        self,
+        cluster_name: str,
+        cluster_role_arn: str,
+        resources_vpc_config: Dict,
+        nodegroup_name: Optional[str] = None,
+        nodegroup_role_arn: Optional[str] = None,
+        compute: Optional[str] = DEFAULT_COMPUTE_TYPE,
+        aws_conn_id: Optional[str] = DEFAULT_CONN_ID,
+        region: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.cluster_name = cluster_name
+        self.cluster_role_arn = cluster_role_arn
+        self.resources_vpc_config = resources_vpc_config
+        self.compute = compute
+        self.aws_conn_id = aws_conn_id
+        self.region = region
+
+        if self.compute == 'nodegroup':
+            self.nodegroup_name = nodegroup_name or self.cluster_name + DEFAULT_NODEGROUP_NAME_SUFFIX
+            if nodegroup_role_arn:
+                self.nodegroup_role_arn = nodegroup_role_arn
+            else:
+                message = "Creating an EKS Managed Nodegroup requires nodegroup_role_arn to be passed in."
+                self.log.error(message)
+                raise AttributeError(message)
```

Comment:
We don't need the log and to raise an error -- by raising the error the message will bubble up anyway, so logging here would just lead to the message appearing twice.

I have also "inverted" the check to have the main flow be "un-intendented" (A _rough_ rule I follow is to try and avoid having code in a block, with the thinking being that the deeper in a code the harder it is to follow.)

```suggestion
            if not nodegroup_role_arn:
                raise ValueError("Creating an EKS Managed Nodegroup requires nodegroup_role_arn to be passed in.")
            self.nodegroup_role_arn = nodegroup_role_arn
```

Reply:
Addressed in https://github.com/apache/airflow/pull/16571/commits/4d59785e903699d74d884a9137197baef3fe0fde